awslabs/aws-lambda-powertools-typescript
# build(logger): reduce node_modules size

## Description of your changes

As detailed in #419 the size of `node_modules` for logger was unnecessarily big, this PR removes the full `lodash` lib in favor of targeted `lodash.x` depenencies brigning down the `node_modules` size to `1.8M` from `139M`.

Detailed list of changes:
- Removed `lodash` & `@types/lodash`
- Added `lodash.clonedeep`, `lodash.merge`, and `lodash.pickby` & respective `@types/lodash.xxx`
- Updated imports to use new deps
- Updated calls for `merge` in main `Logger` class since it was using the `lodash/fp` version [that swaps the argument order](https://www.measurethat.net/Benchmarks/Show/7660/0/lodash-merge-vs-deepmerge-vs-lodashfp-merge).
- Swapped `npm-shrinkwrap.json` for `packages-lock.json` since it was the last utility to still use it.

### How to verify this change

1. Checkout branch & run unit tests in `logger` to check that there are no regressions.
2. Run `npm pack`
3. Install the packed version in a clean project 
4. Run `du -sh *` to measure `node_moduels` size

(see screenshot below for steps 3 & 4)

![Screenshot 2022-01-06 at 20 00 44](https://user-images.githubusercontent.com/7353869/148437711-6160a798-d074-4328-ae50-ef7011d431ba.png)

### Related issues, RFCs

[#419](https://github.com/awslabs/aws-lambda-powertools-typescript/issues/419)  

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist


### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
